### PR TITLE
Fav feature

### PR DESCRIPTION
**Pull request by dimi-23**
**Fixes issue:**

#14

**Changes:**

Added favorites feature.

(1) While searching the stars appear in front of the file names by clicking on which the user can favorite/unfavorite a file: 
image

(2) The favorited files appear at the bottom of the start-up page of the extension:
image

### CODE REVIEW

File: code/popup.html [hunk 1]
```diff
@@ -6,7 +6,22 @@
   <link rel="stylesheet" type="text/css" href="css/bootstrap.css">
   <link rel="stylesheet" type="text/css" href="css/font.css">
   <link rel="shortcut icon" type="image/png" href="/icon/icon.png"/>
-  <title>OpenGenus: World's first Offline Search Engine</title>
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+  <style>
```
Comment (AdiChat) [inline code/popup.html]: Move styling to its own file 👍

File: code/js/popup.js [hunk 2]
```diff
@@ -14,6 +15,24 @@ function AddNewTags (tagName)
 	tags.push(tagName);
 }
 
+function updateFavs(x, filename) {
```
Comment (AdiChat) [inline code/js/popup.js]: Local storage will be deleted when the user will clear his/ her browsing history with cookies and other site data option selected. 

An ideal user may clear history occasionally but we do not want to lose all the data. 

So, you may explore other options like saving in a file in appropriate format.

**Comment by AdiChat** [comment]
Great work 😄 

I have added a few comments to aid you to make this feature the best. Kindly review them and make corresponding changes. 👍

**Comment by dimi-23** [comment]
Hi @AdiChat!

I've made the following  changes as mentioned by you:
- moved styling into its own file
- used Chrome Storage instead of local storage. It solves the problem that you mentioned by using chrome.storage.sync. The stored data will automatically be synced to any Chrome browser that the user is logged into, provided the user has sync enabled. When Chrome is offline, Chrome stores the data locally. The next time the browser is online, Chrome syncs the data. Even if a user disables syncing, storage.sync will still work. In this case, it will behave identically to storage.local.
- Also, since inline functions are no more accepted by Chrome Extensions, replaced them by other functions for the dynamically created elements.

**Comment by AdiChat** [comment]
You must move the javascript and css files to the `js` and `css` folders respectively. 

A lot of files have been included along with the minified versions. You must keep only those that are being used. We must keep the size minimum and not include anything extra. You may, also, edit fontawesome css and keep only the parts that is required. 
You must, also, delete the files `popup_BACKUP_8616.html`, `popup.js.orig`, binary files like `fa-regular-400.woff` and the like.
In short, you must remove all unused files. 👍

**Comment by dimi-23** [comment]
Sorry, I forgot to save space and pushed the repo carelessly. Have made the changes now, you can check them!

**Comment by dimi-23** [comment]
I have made the following changes:

- Corrected the position of stars
- Created a favorites button which opens up favorites in a new page
- UI without any favorites:
image
- UI with favorites:
image

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Comment by dimi-23** [comment]
Yes, am loving it! Thank you so much! 💛
